lib/store.go

```diff
@@ -55,21 +91,42 @@ func (s *store) SetExecutionPayload(blockHash common.Hash, payload *ExecutionPay
 	s.payloadMutex.Lock()
 	defer s.payloadMutex.Unlock()
 
-	s.payloads[blockHash] = payload
+	s.payloads[blockHash] = executionPayloadContainer{payload, time.Now()}
 }
 
 func (s *store) GetForkchoiceResponse(payloadID string) (map[string]string, bool) {
 	s.forkchoiceMutex.RLock()
 	defer s.forkchoiceMutex.RUnlock()
 	forkchoiceResponses, found := s.forkchoices[payloadID]
-	return forkchoiceResponses, found
+	return forkchoiceResponses.Payload, found
 }
 
 func (s *store) SetForkchoiceResponse(boostPayloadID, relayURL, relayPayloadID string) {
 	s.forkchoiceMutex.Lock()
 	defer s.forkchoiceMutex.Unlock()
 	if _, ok := s.forkchoices[boostPayloadID]; !ok {
-		s.forkchoices[boostPayloadID] = make(map[string]string)
+		s.forkchoices[boostPayloadID] = newForkchoiceResponseContainer()
+	}
+	s.forkchoices[boostPayloadID].Payload[relayURL] = relayPayloadID
+}
+
+// Cleanup removes all payloads older than 7 minutes (a bit more than an epoch, which is 6.4 minutes)
```

Comment:
https://github.com/patrickmn/go-cache is another good TTL cache, it's probably overkill for now, maybe for the future : )